.github/workflows/test.yml

```diff
@@ -0,0 +1,45 @@
+name: Tests
+
+on:
+  pull_request:
```

Comment:
We may want to change this to `pull_request_target` so that it will run on PRs from open source contributors, but there are some security considerations: https://securitylab.github.com/research/github-actions-preventing-pwn-requests/.

One thing we can do is update our Actions > General > "Fork pull request workflows from outside collaborators" to "Require approval for all outside collaborators". That way we can take a look at the code before it runs. However, contributors may be able to push and run again after approval.

Maybe there is an alternative to `pull_request_target`, or we can find a way for contributors to run on their fork.

Reply:
I believe the leaving it as `pull_request` will mean we can "approve and run" workflows on external PRs - see https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/approving-workflow-runs-from-public-forks

Our actions are set to require approval:

<img width="897" alt="Screenshot 2024-08-12 at 11 56 48 AM" src="https://github.com/user-attachments/assets/412618ae-e524-4f36-9383-1042caad2b11">